modal/functions.py

```diff
@@ -523,25 +523,22 @@ def _deps():
         fun._spec = class_bound_method._spec
         return fun
 
-    def _hydrate_function_and_method_functions(
-        self, function_id: str, client: _Client, handle_metadata: api_pb2.FunctionHandleMetadata
-    ):
-        self._hydrate(function_id, client, handle_metadata)
+    def _hydrate(self, object_id: str, client: _Client, metadata: Optional[Message]):
```

Comment:
iirc `_hydrate_metadata` is the method you should be overriding in object sub classes to customize how they use the metadata message, without affecting the "base logic" for hydration that could potentially change

Reply:
(but to hydrate an object you still call `._hydrate()` (the object implementation then calls into _hydrate_metadata for the custom logic)

Reply:
Also note this comment in separate PR: https://github.com/modal-labs/modal-client/pull/2510#discussion_r1846186256